### ISSUE TITLE
Authorization for Private Docker Registry does not take effect

### ISSUE THREAD

**Issue by sunny0826** [knqyf263/trivy#60]
**Description**
I used trivy in drone CI and  add TRIVY_AUTH_URL& TRIVY_PASSWORD& TRIVY_USERNAME ,but  it need **authentication required**.

**What did you expect to happen?**
I hope it can pull the image and scan it.

**What happened instead?**

**Output of run with `-debug`:**

```
trivy --exit-code 1 --quiet --severity MEDIUM,HIGH -d registry-vpc.cn-shanghai.aliyuncs.com/keking/kk-devops:prod-25
--
88 | 2019-05-27T03:44:27.336Z	DEBUG	cache dir: /root/.cache/trivy
89 | 2019-05-27T03:44:27.336Z	DEBUG	db path: /root/.cache/trivy/db/trivy.db
90 | 2019-05-27T03:44:27.339Z	INFO	Updating vulnerability database...
91 | 2019-05-27T03:44:27.339Z	DEBUG	git pull
92 | 2019-05-27T03:44:27.339Z	DEBUG	Retrieve the branch being pointed by HEAD
93 | 2019-05-27T03:44:27.339Z	DEBUG	Get the working directory for the repository
94 | 2019-05-27T03:44:27.339Z	DEBUG	Pull the latest changes from the origin remote and merge into the current branch
95 | 2019-05-27T03:44:28.442Z	DEBUG	total updated files: 0
96 | 2019-05-27T03:44:28.592Z	FATAL	error in image scan:
97 | github.com/knqyf263/trivy/pkg.Run
98 | /root/project/pkg/run.go:150
99 | - failed to analyze image:
100 | github.com/knqyf263/trivy/pkg/scanner.ScanImage
101 | /root/project/pkg/scanner/scan.go:29
102 | - failed to extract files:
103 | github.com/knqyf263/fanal/analyzer.Analyze
104 | /go/pkg/mod/github.com/knqyf263/fanal@v0.0.0-20190521154631-a2dde7e171c6/analyzer/analyzer.go:123
105 | - Get https://registry-vpc.cn-shanghai.aliyuncs.com/v2/keking/kk-devops/manifests/prod-25: http: non-successful response (status=401 body="{\"errors\":[{\"code\":\"UNAUTHORIZED\",\"message\":\"authentication required\",\"detail\":[{\"Type\":\"repository\",\"Class\":\"\",\"Name\":\"keking/kk-devops\",\"Action\":\"pull\"}]}]}\n")
```

**Output of `trivy -v`:**

```
trivy version 0.1.1
```

**Additional details (base image name, container registry info...):**
![image](https://user-images.githubusercontent.com/24563928/58394188-8e17d900-8074-11e9-954c-f591e9344cb8.png)

**Comment by tomoyamachi** [comment]
@sunny0826 
We released v0.1.2.
https://github.com/knqyf263/trivy/releases/tag/v0.1.2

Please run again at v0.1.2.

**Comment by tomoyamachi** [comment]
If you have any questions, please reopen this issue.
